streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -151,73 +158,24 @@ public void postCommit() {
         }
     }
 
-    @Override
-    public void prepareCloseClean() {
-        prepareClose(true);
-
-        log.info("Prepared clean close");
-    }
-
-    @Override
-    public void prepareCloseDirty() {
-        prepareClose(false);
-
-        log.info("Prepared dirty close");
-    }
-
-    /**
-     * 1. commit if we are running and clean close;
-     * 2. close the state manager.
-     *
-     * @throws TaskMigratedException all the task has been migrated
-     * @throws StreamsException fatal error, should close the thread
-     */
-    private void prepareClose(final boolean clean) {
-        switch (state()) {
-            case CREATED:
-            case CLOSED:
-                log.trace("Skip prepare closing since state is {}", state());
-                return;
-
-            case RUNNING:
-                if (clean) {
-                    stateMgr.flush();
-                }
-
-                break;
-
-            case RESTORING:
-            case SUSPENDED:
-                throw new IllegalStateException("Illegal state " + state() + " while closing standby task " + id);
-
-            default:
-                throw new IllegalStateException("Unknown state " + state() + " while closing standby task " + id);
-        }
-    }
-
     @Override
     public void closeClean() {
         close(true);
-
         log.info("Closed clean");
     }
 
     @Override
     public void closeDirty() {
         close(false);
-
         log.info("Closed dirty");
     }
 
     @Override
     public void closeAndRecycleState() {
-        prepareClose(true);
+        suspend();
+        prepareCommit();
```

Comment:
Why call `prepareCommit` (or `suspend` for that matter)?

Reply:
Both now do what `prepareClose()` did before.

Reply:
I just mean, why not inline that? I'm just imagining coming back to this code in a few months and wondering why we need to suspend a task before recycling, or why we call `prepareCommit` but don't then actually commit, etc

Reply:
Nevermind, I see that's the pattern we follow elsewhere in this PR